syntax = "proto3";

package UnityRpc;

import "google/protobuf/empty.proto";

enum PlayerType {
	White = 0;
	Black = 1;
}

message Vector2Int {
    int32 x = 1;
    int32 y = 2;
}

message PieceMovement {
	enum PieceType {
		Queen = 0;
		Rook = 1;
		Bishop = 2;
		Knight = 3;
	}
    Vector2Int origin = 1;
    Vector2Int destination = 2;
	bool isEnPassant = 3;
    optional PieceType promotionType = 4;
    bool isCastling = 5;
    string fenString = 6;
	PlayerType currentPlayer = 7;
}

message Result {
    enum StatusType {
        CheckMate = 0;
        Draw = 1;
    }
    StatusType status = 1;
    repeated PlayerType winners = 2;
    repeated PlayerType losers = 3;
}

service UnityRPCBoard {
    rpc Start(google.protobuf.Empty) returns(stream PieceMovement);
}
